### PR TITLE
Kotlin: Fix varargs dataflow, and varargs default handling

### PR DESCRIPTION
**Pull request by smowton**
Dataflow requires accounting for the fact that the varargs parameter isn't necessarily last in the parameter list in a couple more places. Default handling just requires that if the only null parameter is the varargs argument, and it has no default value, then no $default method is required-- the caller is expected to simply pass nothing (at QL / source level) or an empty array (at JVM level).

### CODE REVIEW

File: java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt [hunk 1]
```diff
@@ -15,6 +15,7 @@ import org.jetbrains.kotlin.descriptors.java.JavaVisibilities
 import org.jetbrains.kotlin.ir.IrElement
 import org.jetbrains.kotlin.ir.IrStatement
 import org.jetbrains.kotlin.ir.ObsoleteDescriptorBasedAPI
+import org.jetbrains.kotlin.ir.backend.js.transformers.irToJs.varargParameterIndex
```
Comment (tamasvajk) [inline java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt:18]: This import seems suspicious.

**Review by tamasvajk — COMMENTED** [review]
Found one minor issue, otherwise looks good to me.

File: java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt (hunk 1 above)
Comment (smowton) [reply java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt]: Removed

**Review by tamasvajk — APPROVED** [review]

**Review by aschackmull — APPROVED** [review]
QL changes LGTM.

**Pull request merged**